### DISCUSSION TITLE
How to focus a form input (using refs/exposes)

### DISCUSSION THREAD

**Discussion by okelet** [element-plus/element-plus#15037] [General]
Hi, I am quite new to VueJS and Element Plus, so, please, forgive me if this is a stupid question or has already been answered. I am creating a dialog, that contains a form. I want to focus an input when the dialog is shown. I have tried using:

```html
<el-input v-model="optionsFormValues.name" ref="form_name" />
```

And then:

```js
this.$refs.form_name.focus()
```

But that doesn't work. I have seen that the [component](https://element-plus.org/en-US/component/input.html#exposes) exposes a `ref` attribute, but I couldn't find a way to use it.  Any help?

**Comment by marcgauch** [comment]
Try

````ts
<el-input v-model="optionsFormValues.name" ref="form_name" />


<script setup lang="ts">
import { ref } from 'vue';
const form_name = ref()

function asdf () {
    form_name.value.focus()
}
```

**Comment by Cesar-Huisa** [comment]
This only happened to me before when using el-input inside of a el-dialog, if it's your case you need to execute the code in the "opened" event because it ensures that the function it's executed before the element (in this case the input) renders and avoid calling a non-existent reference that eventually will give you a error.

It should be something like this with the options API:

In methods:

```vue
focus_input() {
  this.$refs['form_name'].focus();
}
``` 

In template:

```vue
<el-dialog  @opened="focus_input" v-model="yourdialogvalue">

    <el-input v-model="optionsFormValues.name" ref="form_name"/>

</el-dialog>
```

**Comment by Murillo-Pinheiro-Sisfrete** [comment]
I was trying to focus on a select element based on a change in another select, but it wasn't working as I expected.

If anyone is in a situation like this, remember to use nextTick();

```vue
  <el-select v-model="..." @change="handleFirstSelectChange"> ... </el-select>
  
  <el-select v-model="..." ref="secondSelectRef"> ... </el-select>
```

```js
const handleFirstSelectChange = async () => {
      await nextTick()
      secondSelectRef.value.focus()
}
```
